src/hw-monitor.h

```diff
@@ -339,6 +339,7 @@ namespace librealsense
 
         virtual std::vector<uint8_t> send( std::vector<uint8_t> const & data ) const;
         virtual std::vector<uint8_t> send( command cmd, hwmon_response * = nullptr, bool locked_transfer = false ) const;
+        virtual std::string get_firmware_version_string(const std::vector<uint8_t>& buff, size_t index, size_t length = 4) const;
```

Comment:
Now that I look at it, I think we were right to no have a template.
However, my opinion is now that that this function does not belong in this class.
I think it should be moved out to the d400- or d500-device implementations. and exposed (if needed) another way.
Still, since I see `get_module_serial_string` is here, I guess they should be kept together. This class needs refactoring...